src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java

```diff
@@ -21,34 +24,119 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
-        if (args.length > 0)
+
+        switch (args.length)
         {
-            if (getPlayer(args[0]) != null)
+            // Mode for seeing how many coins the sender has (doesn't work from console)
+            case 0:
             {
-                p = getPlayer(args[0]);
+                if (senderIsConsole)
+                {
+                    msg("When used from the console, you must define a target player.");
+                }
+                else
+                {
+                    PlayerData playerData = getData(playerSender);
+                    msg(prefix + ChatColor.GREEN + "You have " + ChatColor.RED + playerData.getCoins() + ChatColor.GREEN
+                            + " coins.");
+                }
+                return true;
             }
-            else
+
+            // Mode for seeing how many coins a player has.
+            case 1:
             {
-                msg(PLAYER_NOT_FOUND);
+                Player target = getPlayer(args[0]);
+
+                if (target == null)
+                {
+                    msg(PLAYER_NOT_FOUND);
+                }
+                else
+                {
+                    PlayerData playerData = getData(target);
+                    msg(prefix + ChatColor.GREEN + target.getName() + " has " + ChatColor.RED + playerData.getCoins() + ChatColor.GREEN + " coins.");
+                }
                 return true;
             }
-        }
-        else
-        {
-            if (senderIsConsole)
+
+            // Mode for paying another player coins
+            case 3:
             {
-                msg(prefix + ChatColor.RED + "You are not a player, use /coins <playername>");
-                return true;
+                if (args[0].equalsIgnoreCase("pay"))
+                {
+                    checkPlayer();
+
+                    final Player target = getPlayer(args[1]);
+                    final PlayerData senderData = getData(playerSender);
+
+                    int coinsToTransfer;
+
+                    // Processes args[2]
+                    try
+                    {
+                        // Prevents players from trying to be cheeky with negative numbers.
+                        coinsToTransfer = Math.max(Math.abs(Integer.parseInt(args[2])), 1);
+                    }
+                    catch (NumberFormatException ex)
+                    {
+                        msg("Invalid number: " + args[2], ChatColor.RED);
+                        return true;
+                    }
+
+                    // Prevents players from performing transactions they can't afford to do.
+                    if (senderData.getCoins() < coinsToTransfer)
+                    {
+                        msg("You don't have enough coins to perform this transaction.", ChatColor.RED);
+                        return true;
+                    }
+
+                    if (target == null)
+                    {
+                        msg(PLAYER_NOT_FOUND);
```

Comment:
Do we not also need to return true out of this statement?

Reply:
Already a thing at line 117